src/config/id.rs

```diff
@@ -13,6 +13,15 @@ pub enum ComponentScope {
     Pipeline(String),
 }
 
+impl fmt::Display for ComponentScope {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Global => write!(f, "global"),
+            Self::Pipeline(name) => write!(f, "pipeline:{}", name),
```

Comment:
Should this `:` be a `.`? I thought we were joining the names with `.`'s now?

Reply:
This is to display the scope, it's not related to the `ComponentKey` itself.
With this, in the events from `internal_scope`, you'll have `{ "component_id": "foo", "component_scope": "pipeline:bar" }` so that when we introduce more scopes, we can just prefix it with the type.
Besides that, I'm ok to use `.` instead of `:`. In the previous PR, nobody raised this point (maybe nobody noted this 🤷 ) so I guess @lukesteensen and @jszwedko are ok with that 🤔

Reply:
Cool - I wasn't sure if the event tagging would match like `"component_scope": "pipeline.bar"`

Reply:
The colon seems reasonable to me since it's not the name itself but the type of the scope, but I don't have a particularly strong opinion.